lib/models/device.coffee

```diff
@@ -1917,6 +1917,170 @@ getDeviceModel = (deps, opts) ->
+				return should_be_running__release[0].commit
+			belongs_to__application[0].commit
+		.asCallback(callback)
+
+	###*
+	# @summary Set a specific device to run a particular release
+	# @name setToRelease
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @description Configures the device to run a particular release
+	# and not get updated when the current application release changes.
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @param {String} fullReleaseHash - the hash of a successful release (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.setToRelease('7cf02a6, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.device.setToRelease(123, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.device.setToRelease('7cf02a6, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847', function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.setToRelease = (uuidOrId, fullReleaseHashOrId, callback) ->
+		Promise.try ->
+			if isId(uuidOrId) and isId(fullReleaseHashOrId)
+				return {
+					deviceId: uuidOrId
+					releaseId: fullReleaseHashOrId
+				}
+
+			releaseFilterProperty = if isId(fullReleaseHashOrId) then 'id' else 'commit'
+			exports.get(uuidOrId,
+				$select: 'id'
+				$expand:
+					belongs_to__application:
+						$select: 'id'
+						$expand:
+							owns__release:
+								$top: 1
+								$select: 'id'
+								$filter:
+									"#{releaseFilterProperty}": fullReleaseHashOrId
+								# if there is a successful release for a specific commit hash
+								# then it will be the last created
+								$orderby: 'created_at desc'
+			)
+			.then ({ id, belongs_to__application }) ->
+				app = belongs_to__application[0]
+				release = app.owns__release[0]
+				if not release
+					throw new errors.ResinReleaseNotFound(fullReleaseHashOrId)
+				return {
+					deviceId: id
+					releaseId: release.id
+				}
+		.then ({ deviceId, releaseId }) ->
+			pine.patch
+				resource: 'device'
+				id: deviceId
+				body: should_be_running__release: releaseId
+		.asCallback(callback)
+
+	###*
+	# @summary Configure a specific device to track the current application release
+	# @name enableApplicationReleaseTracking
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @description The device's current release will be updated with each new successfully built release.
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.enableApplicationReleaseTracking('7cf02a6).then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.device.enableApplicationReleaseTracking('7cf02a6, function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.enableApplicationReleaseTracking = (uuidOrId, callback) ->
```

Comment:
As in your comment - I do think renaming this to `trackApplicationRelease` (and the app equivalent to `trackLatestRelease`) is a good plan.